internal/pkg/otel/Makefile

```diff
@@ -0,0 +1,6 @@
+.PHONY: all
+
+all:
+	go run go.opentelemetry.io/collector/cmd/builder@v0.107.0 --skip-compilation --skip-get-modules --config builder-config.yaml
+	@rm main.go main_others.go main_windows.go go.mod
+	sed -i 's/package main/\/\/go:build \!windows\npackage otel/g' components.go
```

Comment:
I haven't looked into why there's a `!windows` build flag, but I would like to understand the reason behind it.

Reply:
Having otel dependencies breaks agent on startup as a Windows service. Lee is working on finding out what's causing this precisely

Reply:
More details here: https://github.com/elastic/elastic-agent/issues/4976